### PR TITLE
Created StepAction tekton-catalog-publish

### PR DESCRIPTION
**Pull request by chitrangpatel**
Added a StepAction `tekton-catalog-publish` to assist in publishing the StepAction catalog as a bundle.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [x] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages
/kind misc

### CODE REVIEW

File: stepaction/tekton-catalog-publish/0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+Bundles are published to `$REGISTRY/$PATH/<task/stepaction-name>:<task/stepaction-version>` and, when `TAG` is provided, to
+`$REGISTRY/$PATH/<task/stepaction-name>:$TAG`. An example of extra tag could be the git sha of the catalog repo that
+is being published. The task uses the `tkn bundle` command to publish, which is available
+in `tkn` starting with version v0.18.0.
+
+## Install the StepAction
+
+```bash
+kubectl apply -f
```
Comment (vinamra28) [inline stepaction/tekton-catalog-publish/0.1/README.md:14]: can we complete this? 😅

File: stepaction/tekton-catalog-publish/0.1/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# Tekton Catalog Publish
+
+This stepaction publishes each Task/StepAction in a Tekton catalog as [Tekton Bundles](https://tekton.dev/docs/pipelines/pipelines/#tekton-bundles).
+The catalog must be structured as a [Tekton Catalog](https://github.com/tektoncd/catalog#catalog-structure).
+
+Bundles are published to `$REGISTRY/$PATH/<task/stepaction-name>:<task/stepaction-version>` and, when `TAG` is provided, to
+`$REGISTRY/$PATH/<task/stepaction-name>:$TAG`. An example of extra tag could be the git sha of the catalog repo that
+is being published. The task uses the `tkn bundle` command to publish, which is available
+in `tkn` starting with version v0.18.0.
```
Comment (vinamra28) [inline stepaction/tekton-catalog-publish/0.1/README.md:9]: we should hold on right now with this as currently tkn doesn't support stepactions
cc @piyush-garg @vdemeester

File: stepaction/tekton-catalog-publish/0.1/tests/resources.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: catalog-source-pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 500Mi
```
Comment (vinamra28) [inline stepaction/tekton-catalog-publish/0.1/tests/resources.yaml:11]: do we need 500Mi for this test? 😅

File: stepaction/tekton-catalog-publish/0.1/README.md (hunk 2 above)
Comment (chitrangpatel) [reply stepaction/tekton-catalog-publish/0.1/README.md:9]: I think we just need the `tkn bundle push` command here. I don't think it cares what is in the bundle (at least I don't think so) which is why the CI tests pass. So technically to run this command, support for Step actions is not needed. I might be completely wrong here though.

File: stepaction/tekton-catalog-publish/0.1/tests/resources.yaml (hunk 3 above)
Comment (chitrangpatel) [reply stepaction/tekton-catalog-publish/0.1/tests/resources.yaml:11]: This was borrowed from the Task. For now I don't think so. Eventually StepActions will grow to the same size as Tasks. 

Actually, we could get by with an emptyDir here. Let me do that actually.

File: stepaction/tekton-catalog-publish/0.1/README.md (hunk 1 above)
Comment (chitrangpatel) [reply stepaction/tekton-catalog-publish/0.1/README.md:14]: Oops. Yes, I can fix this.

**Review by vinamra28 — APPROVED** [review]

**Pull request merged**